Relationships/RelationshipResolver.cs

```diff
@@ -70,49 +71,93 @@ public class RelationshipResolver
         // A list of all the mods we're going to install.
         private static readonly ILog log = LogManager.GetLogger(typeof (RelationshipResolver));
         private readonly Dictionary<string, CkanModule> modlist = new Dictionary<string, CkanModule>();
-
-        private readonly List<KeyValuePair<Module, Module>> conflicts =
+        private readonly List<CkanModule> user_requested_mods = new List<CkanModule>();
+        private readonly List<KeyValuePair<Module, Module>> conflicts = 
             new List<KeyValuePair<Module, Module>>();
+        private readonly Dictionary<Module, Relationship> reasons = 
+            new Dictionary<Module, Relationship>(new Module.IdentifierEqualilty());
 
-        private readonly Dictionary<Module, Relationship> reasons = new Dictionary<Module, Relationship>();
         private readonly Registry registry;
         private readonly KSPVersion kspversion;
+        private readonly RelationshipResolverOptions options;
+        private readonly HashSet<Module> installed_modules;
+
+
+        public RelationshipResolver(RelationshipResolverOptions options, Registry registry, KSPVersion kspversion)
```

Comment:
A wild Constructor appears!

So, this is creating a RR and gives it a registry, but no modules to install? That seems odd, and the documentation is missing. Let's look at the code...

It seems like we're grabbing all the installed modules from the registry, and then marking them all as pre-installed. I'm guessing it's the responsibility of the caller to invoke `AddModulesToInstall` after creating the object?